### ISSUE TITLE
terminal emulators may fail to build on MacOS

### ISSUE THREAD

**Issue by gwright83** [aap/vt05#20]
The terminal emulator for the vt05, vt52, dp3300, dm2500 and gecon may fail to build on MacOS with `libsdl2`.  This was discovered building ITS using the pdp10-kl emulator.  MacOS was 11.6.4, libsdl2 version 2.0.20 installed by MacPorts and clang13.

The error was
```
cc -Wall -Wno-unused-parameter -Wno-parentheses -Wno-unused-result -O3 -o vt05 vt05.c terminal.c `sdl2-config --libs --cflags` -lm -lpthread
In file included from vt05.c:18:
In file included from /opt/local/include/SDL2/SDL.h:32:
In file included from /opt/local/include/SDL2/SDL_main.h:25:
/opt/local/include/SDL2/SDL_stdinc.h:482:5: error: implicit declaration of function 'memset_pattern4' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    memset_pattern4(dst, &val, dwords * 4);
    ^
1 error generated.
make: *** [vt05] Error 1
```
The cause of the error seems to be fragile conditionalization in the `SDL.h` header file.  The error can be fixed my moving
the include of `SDL.h` to the top of the includes (for example, here in vt05.c):
```
#define _XOPEN_SOURCE 600
#define _DEFAULT_SOURCE

#include <SDL.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <stdarg.h>
#include <ctype.h>
```
Making this change in all of the source files allow the build to complete successfully.

The fragile conditionalization in `SDL.h` seems to be a known problem and the recommended solution is to include `SDL.h`
as early as possible.

**Comment by larsbrinkhoff** [comment]
I made that change to all .c files on a branch: https://github.com/aap/vt05/tree/lars/move-sdlh

I don't have a mac to test this.  @gwright83 can you check that my branch fixes the compile error?

**Comment by larsbrinkhoff** [comment]
![bugfix](https://user-images.githubusercontent.com/775050/156711018-129213a5-7b78-4837-84fb-c96ccefccbab.jpg)

**Comment by gwright83** [comment]
I tested the fix by building ITS for pdp10-kl on MacOS using the updated submodules and it ran successfully.
I also did a complete build of ITS for the pdp10-ka, -kl, simh and and klh10 emulators on FreeBSD (my CI machine).  Those ran to completion successfully as well.

The changes appear to have fixed the problem.  Thank you.

However,
![lack_of_faith](https://user-images.githubusercontent.com/3758632/156779459-282a55da-445e-4635-ae83-2a05578f3617.jpeg)
